### ISSUE TITLE
Singularity Pull Docker Image From private repository error

### ISSUE THREAD

**Issue by ortizem** [apptainer/singularity#1322]
## Version of Singularity:
2.4-dist
Write here.

### Expected behavior

 singularity pull docker://docker.repo.com:80/nm74-centos7:latest

-bash-4.2$ singularity pull docker://docker.repo.com:80/nm74-centos7:latest
WARNING: pull for Docker Hub is not guaranteed to produce the
WARNING: same image on repeated pull. Use Singularity Registry
WARNING: (shub://) to pull exactly equivalent images.
Traceback (most recent call last):
  File "/usr/libexec/singularity/python/import.py", line 109, in <module>
    main()
  File "/usr/libexec/singularity/python/import.py", line 86, in main
    layerfile=LAYERFILE)
  File "/usr/libexec/singularity/python/docker/main.py", line 83, in IMPORT
    client = DockerApiConnection(image=image, auth=auth)
  File "/usr/libexec/singularity/python/docker/api.py", line 89, in __init__
    self.load_image(kwargs['image'])
  File "/usr/libexec/singularity/python/docker/api.py", line 141, in load_image
    self.update_token()
  File "/usr/libexec/singularity/python/docker/api.py", line 152, in update_token
    response = self.get_tags(return_response=True)
  File "/usr/libexec/singularity/python/docker/api.py", line 271, in get_tags
    return_response=return_response)
  File "/usr/libexec/singularity/python/base.py", line 262, in get
    return_response=return_response)
  File "/usr/libexec/singularity/python/base.py", line 278, in submit_request
    response = urlopen(request)
  File "/usr/lib64/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python2.7/urllib2.py", line 431, in open
    response = self._open(req, data)
  File "/usr/lib64/python2.7/urllib2.py", line 449, in _open
    '_open', req)
  File "/usr/lib64/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 1258, in https_open
    context=self._context, check_hostname=self._check_hostname)
  File "/usr/lib64/python2.7/urllib2.py", line 1214, in do_open
    raise URLError(err)
urllib2.URLError: <urlopen error [SSL: UNKNOWN_PROTOCOL] unknown protocol (_ssl.c:579)>
Cleaning up...

Write here.

singularity pull docker://docker.repo.com:80/nm74-centos7:latest

I am getting an error on pulling a Docker image from a private repository. Do you have a workaround this problem? Please advise.

**Comment by vsoch** [comment]
Try without https.

```
SINGULARITY_NOHTTPS=yes singularity pull docker://docker.repo.com:80/nm74-centos7:latest
```
Why do you need to specify port 80? I would try removing that too - 80 is standard.

**Comment by ortizem** [comment]
Hi @vsoch 
Thank you, it worked! But now I get a "Permission Denied" error. Is that something I need to do on my Singularity side? 

-bash-4.2$ SINGULARITY_NOHTTPS=yes singularity pull docker://docker.repo.com/nm74-centos7:latest
WARNING: pull for Docker Hub is not guaranteed to produce the
WARNING: same image on repeated pull. Use Singularity Registry
WARNING: (shub://) to pull exactly equivalent images.
Docker image path: docker.repo.com/nm74-centos7:latest
Cache folder set to /home/user/.singularity/docker
[13/13] |===================================| 100.0%
Importing: base Singularity environment
Importing: /home/user/.singularity/docker/sha256:af4b0a2388c69010cf675c050e51cb1fabbdf2303f955c31805b280324fd4523.tar.gz
Importing: /home/user/.singularity/docker/sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4.tar.gz
Importing: /home/user/.singularity/docker/sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4.tar.gz
Importing: /home/user/.singularity/docker/sha256:a3ed95caeb02ffe68cdd9fd84406680ae93d633cb16422d00e8a7c22955b46d4.tar.gz
Importing: /home/user/.singularity/docker/sha256:52b6fefd3aaa7e67cf7726e7fc43996671d3864b94a5d4853a6d5b79991f9e70.tar.gz
tar: usr/bin/a2p: Cannot open: Permission denied
tar: usr/bin/aclocal: Cannot open: Permission denied
tar: usr/bin/aclocal-1.13: Cannot hard link to `usr/bin/aclocal': No such file or directory
tar: usr/bin/autoconf: Cannot open: Permission denied
tar: usr/bin/autoheader: Cannot open: Permission denied
tar: usr/bin/autom4te: Cannot open: Permission denied
tar: usr/bin/automake: Cannot open: Permission denied
tar: usr/bin/automake-1.13: Cannot hard link to `usr/bin/automake': No such file or directory

**Comment by vsoch** [comment]
Yes, obtaining the layers is indication that communication with your local registry is working. The tar errors are a different issue related to singularity, I'll let other pick up here!

**Comment by dtrudg** [comment]
Hi @ortizem - are you able to test this with the `testing-2.4.3` branch? There are fixes there in testing which I'm hoping will address this issue for you.

**Comment by dtrudg** [comment]
Hi @vsoch - going to self-assign this now to track what seems to now be a whiteout issue. Cheers!

**Comment by vsoch** [comment]
Cheers!

**Comment by dtrudg** [comment]
Closing as extraction fixes which include addressing the errors report here were included in 2.5.0. Please feel to reopen if you experience further issues.
